examples/grouped_gemm.py

```diff
@@ -0,0 +1,330 @@
+    """
+    assert len(group_A) == len(group_B), "group_A and group_B must have same length"
+    group_size = len(group_A)
+
+    if group_size == 0:
+        return []
+
+    group_C = []
+
+    # Process each group with a separate kernel call
+    # This allows helion to optimize each GEMM individually
+    for i in range(group_size):
+        A = group_A[i]
+        B = group_B[i]
+
+        # Call the optimized single GEMM kernel
+        C = single_gemm(A, B)
+        group_C.append(C)
+
+    return group_C
+
+
+# Alternative implementation using concatenated tensors (more advanced)
+@helion.kernel(static_shapes=False)
+def grouped_gemm_concatenated(
+    A_concat: torch.Tensor,  # Concatenated A matrices
+    B_concat: torch.Tensor,  # Concatenated B matrices
+    A_offsets: torch.Tensor,  # [group_size + 1] - Start indices for A matrices
+    B_offsets: torch.Tensor,  # [group_size + 1] - Start indices for B matrices
+    C_offsets: torch.Tensor,  # [group_size + 1] - Start indices for C matrices
+    group_shapes: torch.Tensor,  # [group_size, 3] - [M, N, K] for each group
+    group_size: int,
+) -> torch.Tensor:
+    """
+    Alternative implementation using concatenated tensors to avoid list indexing.
+    This version processes all groups in a single kernel launch.
+    """
+    # Calculate total output size
+    total_output_size = C_offsets[group_size].item()
+    C_concat = torch.empty(
+        [total_output_size], device=A_concat.device, dtype=A_concat.dtype
+    )
+
+    # Process each group
+    for g in range(group_size):
+        # Get dimensions for this group
+        M = group_shapes[g, 0].item()
+        N = group_shapes[g, 1].item()
+        K = group_shapes[g, 2].item()
+
+        # Skip empty groups
+        if M == 0 or N == 0 or K == 0:
+            continue
+
+        # Get offsets for this group
+        a_start = A_offsets[g].item()
+        a_end = A_offsets[g + 1].item()
+        b_start = B_offsets[g].item()
+        b_end = B_offsets[g + 1].item()
+        c_start = C_offsets[g].item()
+        # c_end = C_offsets[g + 1].item()
+
+        # Extract matrices for this group
+        A_flat = A_concat[a_start:a_end]
+        B_flat = B_concat[b_start:b_end]
+
+        # Reshape to proper matrix shapes
+        A = A_flat.view(M, K)
+        B = B_flat.view(K, N)
+
+        # Compute matrix multiplication for this group
+        for tile_m, tile_n in hl.tile([M, N]):
+            acc = hl.zeros([tile_m, tile_n], dtype=torch.float32)
+
+            for tile_k in hl.tile(K):
+                acc = torch.addmm(acc, A[tile_m, tile_k], B[tile_k, tile_n])
+
+            # Store result in concatenated output
+            C_group = acc.view(-1)  # Flatten the result
+            tile_start = c_start + tile_m.begin * N + tile_n.begin
+            tile_end = tile_start + C_group.numel()
+            C_concat[tile_start:tile_end] = C_group
```

Comment:
I'm a bit surprised this works.  Is this tested?

Reply:
Yes it works, I followed the same shape inputs as the triton grouped gemm example in the file examples/grouped_gemm.py.